File: c#/src/Sodium/Sodium/MutableMaybeValue.cs
using System;

namespace Sodium
{
    public class MutableMaybeValue<T>
    {
        public Maybe<T> Value { get; private set; }

        public void Set(T value) => this.Value = Maybe.Some(value);
        public void Reset() => this.Value = Maybe.None;

        public void Match(Action<T> hasValueAction, Action nothingAction) => this.Value.Match(hasValueAction, nothingAction);
        public TResult Match<TResult>(Func<T, TResult> hasValueFunc, Func<TResult> nothingFunc) => this.Value.Match(hasValueFunc, nothingFunc);
    }
}

File: swift/src/Sodium/Sodium/Sodium/Sodium/MutableMaybeValue.swift
public class MutableMaybeValue<T>
{
    private var value = Maybe.Nothing<T>()

    public func Set(value: T) { self.value = Maybe.Just(value) }
    public func Reset() { self.value = Maybe.Nothing<T>() }
    public func Get() -> IMaybe<T> { return self.value }
}
